src/Uno.UI/UI/Xaml/Controls/Control/Control.cs

```diff
@@ -293,6 +293,7 @@ internal View TemplatedRoot
 						}
 						else
 						{
+							_applyTemplateShouldBeInvoked = false;
```

Comment:
Which scenario does this change fix? If it does not break any tests, let's add some new ones.

Reply:
Here's the problematic scenario that happens with Fluent-theme Sliders in Gallery. Here I'm using ViewBoxSamplePage as an example of a Gallery page with a slider.

1. `ViewBoxSamplePage.CreationComplete()` -> `ViewBoxSamplePage.ApplyStyles()` -> ... ->`Slider.OnTemplateChanged()` -> ... -> `Slider.set_TemplatedRoot` -> the slider hasn't been loaded yet so we get _applyTemplateShouldBeInvoked = true.
2. `ViewBoxSamplePage` is added to the visual tree -> `ViewBoxSamplePage.OnElementLoading()` -> ... ->`Slider.OnElementLoading()` -> `Slider.OnFwEltLoading()` -> `Slider.OnLoadingPartial()` -> `Slider.ApplyStyles()` -> ... -> `Slider.OnTemplateChanged()` -> ... -> `Slider.set_TemplatedRoot` -> first OnApplyTemplate()
3. `Slider.OnPostLoading()` -> `Slider.TryCallOnApplyTemplate()` -> _applyTemplateShouldBeInvoked was never reset, so OnApplyTemplate() fires again.

Slider.OnApplyTemplate() is written in a way that makes repeated calls destructive (it subscribes to pointer events and then sets a SerialDisposable to unsubscribe those same events, so if OnApplyTemplate is called a second time, it will (idempotently) resubscribe and then unsubscribe from pointer events, which causes the problem on Gallery.

I've tried to write a test for this but I couldn't reproduce something inside SamplesApp. Probably because I'm missing the first step? Not sure.

Reply:
Could it be caused by the fact that the fluent theme is not enabled? The default theme in runtime tests is the old UWP, and you can enable it selectively there. Otherwise, You may be able to create a test that explicitly tests for the behavior changed by that property, but loading/unloading the control at appropriate times?